### PR TITLE
fix: bank transactions - monthly view - use date range in api queries

### PR DESCRIPTION
**Pull request by tomaszantas**
## Description

Use filter's `dateRange` when bank transactions is shown in the monthly view.

## How this has been tested?

The history could be cut off after few months because only first 200 records were fetched.
For the `8a...4ace` business, the history of categorized cut off on March, so February was empty:

image

After fix:

image

### CODE REVIEW

File: src/hooks/useIsVisible/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export { useIsVisible } from './useIsVisible'
```
Comment (imkesin) [inline src/hooks/useIsVisible/index.ts:1]: Nit: I'd like us to [stop using barrel files](https://tkdodo.eu/blog/please-stop-using-barrel-files).

File: src/hooks/useBankTransactions/useBankTransactions.tsx [hunk 2]
```diff
@@ -377,6 +394,12 @@ export const useBankTransactions: UseBankTransactions = params => {
       })
   }
 
+  useEffect(() => {
+    if (monthlyView && lastMetadata?.pagination?.has_more) {
+      fetchMore()
+    }
+  }, [lastMetadata])
+
```
[the remark this answers is not in the document]
Comment (tomaszantas) [reply src/hooks/useBankTransactions/useBankTransactions.tsx]: Yeah, I agree and I already changed it. In the new approach, I removed this `useEffect` and new pages are being fetched when user scrolls to the end of the list.

File: src/hooks/useIsVisible/index.ts (hunk 1 above)
Comment (tomaszantas) [reply src/hooks/useIsVisible/index.ts:1]: Make sense to me. Which approach do you prefer:
1. `utils/useIsVisible/useIsVisible.ts`,
2. `utils/useIsVisible/index.ts`,
3. `utils/useIsVisible.ts`?

I would go with the last one, and when we need to split logic into more files, then [2]. What do you think?
I would make such changes in batch after merging this and eslint related pull requests.

Comment (imkesin) [reply src/hooks/useIsVisible/index.ts:1]: I like the third approach. 

As soon as we get an unreasonable number of files at the same level, then we can group a few into a folder.

**Review by imkesin — APPROVED** [review]

**Pull request merged**